src/ray/gcs/gcs_server/gcs_actor_manager.h

```diff
@@ -346,6 +345,17 @@ class GcsActorManager : public rpc::ActorInfoHandler {
   /// \param actor The actor to be killed.
   void AddDestroyedActorToCache(const std::shared_ptr<GcsActor> &actor);
 
+  std::shared_ptr<rpc::ActorTableData> GenActorDataOnlyWithStates(
+      const rpc::ActorTableData &actor) {
+    auto actor_delta = std::make_shared<rpc::ActorTableData>();
+    actor_delta->set_state(actor.state());
+    actor_delta->mutable_address()->CopyFrom(actor.address());
+    actor_delta->set_num_restarts(actor.num_restarts());
+    actor_delta->set_timestamp(actor.timestamp());
+    actor_delta->set_pid(actor.pid());
+    return actor_delta;
+  }
```

Comment:
Is there any way to maintain key consistency between the Ray core and dashboard?

Reply:
It's hard to do this, as ways of setting value for fields is totally different in cpp and python.

Reply:
Maybe we can add some comments in cpp codes, and point the comment to some constants in python.

Reply:
We should just add an unit test to detect this.

Reply:
@WangTaoTheTonic Can you make sure to add unit tests that can detect inconsistency here?

Reply:
ok that's a good idea. I'll check.